vms/proposervm/vm.go

```diff
@@ -220,6 +227,34 @@ func (vm *VM) Initialize(
 	default:
 		return err
 	}
+
+	vm.proposerBuildSlotGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "block_building_slot",
+			Help: "the post-durango slot in which the block was built",
+		})
+
+	if err = registerer.Register(vm.proposerBuildSlotGauge); err != nil {
+		return err
+	}
+
+	vm.acceptedBlocksSlotHistogram = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Name: "accepted_blocks_slot_histogram",
+			Help: "the post-durango slot in which the block was accepted at",
+			// define the following ranges:
+			// (-inf, 0]
+			// (0, 1]
+			// (1, 2]
+			// (2, inf)
+			// the usage of ".5" before was to ensure we work around the limitation
+			// of comparing floating point of the same numerical value.
+			Buckets: []float64{0.5, 1.5, 2.5},
+		},
+	)
+	if err = registerer.Register(vm.acceptedBlocksSlotHistogram); err != nil {
+		return err
+	}
 	return nil
```

Comment:
We typically use `utils.Err` for checks like this to improve readability.
```suggestion
	vm.proposerBuildSlotGauge = prometheus.NewGauge(prometheus.GaugeOpts{
		Name: "block_building_slot",
		Help: "the slot that this node may attempt to build a block",
	})
	vm.acceptedBlocksSlotHistogram = prometheus.NewHistogram(prometheus.HistogramOpts{
		Name: "accepted_blocks_slot",
		Help: "the slot accepted blocks were proposed in",
		// define the following ranges:
		// (-inf, 0]
		// (0, 1]
		// (1, 2]
		// (2, inf)
		// the usage of ".5" before was to ensure we work around the limitation
		// of comparing floating point of the same numerical value.
		Buckets: []float64{0.5, 1.5, 2.5},
	})
	return utils.Err(
		registerer.Register(vm.proposerBuildSlotGauge),
		registerer.Register(vm.acceptedBlocksSlotHistogram),
	)
```

Reply:
that one is really nice. I haven't seen that design pattern used before.. and I like it.